internal/app/machined/pkg/controllers/k8s/kubelet_service.go

```diff
@@ -196,6 +197,22 @@ func (ctrl *KubeletServiceController) Run(ctx context.Context, r controller.Runt
 			}
 		}
 
+		policy, err := getFromMap[string](cfgSpec.Config, "cpuManagerPolicy")
+		if err != nil {
+			return err
+		}
+
+		newPolicy := policy.ValueOrZero()
```

Comment:
this is a wrong way to detect the change, as it only works if the controller was up when the configuration change happened, while the controller might be down on config change (e.g. applying machine config with a reboot).

we should rather detect the policy name in the `state` file doesn't match the policy in the config.

Reply:
Done here: https://github.com/siderolabs/talos/pull/7557/files#diff-fa70ddd6c68e2179d952fc464512dd777494c4fe1d53cce9940c2c5c20ed9183R234-R254